modules/cockroachdb/options.go

```diff
@@ -67,3 +69,26 @@ func WithTLS(cfg *TLSConfig) Option {
 		o.TLS = cfg
 	}
 }
+
+// DefaultStatements are the settings recommended by Cockroach Labs for testing clusters.
+// Note that to use these defaults the user needs to have MODIFYCLUSTERSETTING privilege.
+// See https://www.cockroachlabs.com/docs/stable/local-testing for more information.
+var DefaultStatements = []string{
+	"SET CLUSTER SETTING kv.range_merge.queue_interval = '50ms'",
+	"SET CLUSTER SETTING jobs.registry.interval.gc = '30s'",
+	"SET CLUSTER SETTING jobs.registry.interval.cancel = '180s'",
+	"SET CLUSTER SETTING jobs.retention_time = '15s'",
+	"SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false",
+	"SET CLUSTER SETTING kv.range_split.by_load_merge_delay = '5s'",
+	`ALTER RANGE default CONFIGURE ZONE USING "gc.ttlseconds" = 600`,
+	`ALTER DATABASE system CONFIGURE ZONE USING "gc.ttlseconds" = 600`,
+}
+
+// WithStatements sets the statements to run on the CockroachDB cluster once the container is ready.
+// This, in combination with DefaultStatements, can be used to configure the cluster with the settings
+// recommended by Cockroach Labs.
```

Comment:
suggestion: This is somewhat confusing as this won't be combined with `DefaultStatements` which is how I would interpret this currently. We should clarify and ensure we clearly state the default if not configured is `DefaultStatements`.

Reply:
If we are always adding the default statements, then I think they must be private.

Reply:
> If we are always adding the default statements, then I think they must be private, and probably prepended to the user-provided ones.

I see what you mean, but given the complications with TLS and users without MODIFYCLUSTERSETTING privilege I think prepending to the user-provided statements is probably going to cause a bit of pain.

Good spot on the comment. I'll hold on that change until we've established what we do about the TLS and user privileges thing though, as it might affect this.

Reply:
Private vs public for these is an interesting discussion, here's some ideas from the perspective of making the var private:
* Pro: Others can't mess with them
* Conn: Others can't correct if needed
* Conn: Can't pass them into WithStatements to customise
* Conn: Not self documenting.

Reply:
One of the goals of any testcontainers module is to be simple, and maybe opinionated, helping users in not committing mistakes. So I'd look for the mechanism that creates this experience.

Whatever you find that matches that, it's fine for me 😊